### PR TITLE
manage upload of big size files - pn-7268

### PR DESCRIPTION
**Pull request by AndreaCimini90**
## Short description
In New notification page, step Attachments, front-end freezes if we try to upload big files. 

## List of changes proposed in this pull request
- Mooved the calculation of Unit8Array to the api call

## How to test
Go to new notification page, upload a big file and check that the notification creation goes well (try the file in the linked issue)

### CODE REVIEW

**Review by clombardi — CHANGES_REQUESTED** [review]
In general OK, and in fact I found no bugs, just some details that could be cleaned.

File: packages/pn-commons/src/components/FileUpload.tsx [hunk 1]
```diff
@@ -14,32 +14,21 @@ import AttachFileIcon from '@mui/icons-material/AttachFile';
 import CloseIcon from '@mui/icons-material/Close';
 import InfoOutlinedIcon from '@mui/icons-material/InfoOutlined';
 import { ButtonNaked } from '@pagopa/mui-italia';
-import {
-  calcUnit8Array,
-  calcSha256String,
-  calcBase64String,
```
[the remark this answers is not in the document]
Comment (AndreaCimini90) [reply packages/pn-commons/src/components/FileUpload.tsx:20]: Yes! I commented why

File: packages/pn-pa-webapp/src/pages/components/NewNotification/Attachments.tsx [hunk 2]
```diff
@@ -189,6 +182,20 @@ const Attachments = ({
     []
   );
 
+  const storeAttachments = (documents: Array<NewNotificationDocument>) => {
+    dispatch(
+      setAttachments({
+        documents: documents.map((v) => ({
+          ...v,
+          file: {
```
[the remark this answers is not in the document]
Comment (AndreaCimini90) [reply packages/pn-pa-webapp/src/pages/components/NewNotification/Attachments.tsx:190]: It is a mistake

File: packages/pn-pa-webapp/src/redux/newNotification/actions.ts [hunk 3]
```diff
@@ -75,9 +80,13 @@ export const uploadNotificationAttachment = createAsyncThunk<
   async (items: Array<NewNotificationDocument>, { rejectWithValue }) => {
     try {
       // before upload, filter out documents already uploaded
-      const itemsToUpload = items
-        .filter((item) => !item.ref.key && !item.ref.versionToken)
-        .map((item) => createPayloadToUpload(item));
+      const itemsToUpload = [];
+      const filteredItems = items.filter((item) => !item.ref.key && !item.ref.versionToken);
+      for (const item of filteredItems) {
```
[the remark this answers is not in the document]
Comment (AndreaCimini90) [reply packages/pn-pa-webapp/src/redux/newNotification/actions.ts:85]: You're right! we can use Promise.all

File: packages/pn-pa-webapp/src/redux/newNotification/actions.ts [hunk 4]
```diff
@@ -100,23 +109,25 @@ export const uploadNotificationAttachment = createAsyncThunk<
...
 );
 
-const getPaymentDocumentsToUpload = (items: {
+const getPaymentDocumentsToUpload = async (items: {
   [key: string]: PaymentObject;
-}): Array<UploadDocumentParams> =>
-  Object.values(items).reduce((arr, item) => {
+}): Promise<Array<UploadDocumentParams>> => {
+  const documentsArr: Array<UploadDocumentParams> = [];
+  for (const item of Object.values(items)) {
```
[the remark this answers is not in the document]
Comment (AndreaCimini90) [reply packages/pn-pa-webapp/src/redux/newNotification/actions.ts:116]: I used the Promise.all method

[the remark this answers is not in the document]
Comment (clombardi) [reply packages/pn-pa-webapp/src/redux/newNotification/actions.ts:111]: great!

**Review by clombardi — APPROVED** [review]
OK, ready to merge

**Pull request merged**